zap/src/output/luau/client.rs

```diff
@@ -366,27 +435,39 @@ impl<'src> ClientOutput<'src> {
 				self.push_line(&format!("events[{id}] = {{}}"));
 			}
 
+			if ev_decl.data.is_some() {
+				self.push_line(&format!("event_queue[{id}] = {{}}"));
+			} else {
+				self.push_line(&format!("event_queue[{id}] = 0"));
+			}
+		}
+
+		for (i, ..) in self.config.fndecls.iter().enumerate() {
+			let id = i + 1;
 			self.push_line(&format!("event_queue[{id}] = {{}}"));
```

Comment:
we should replace `{}` with `table.create(255)` to avoid resizing costs

Reply:
but won't that use extra memory? or is that just not how it works

Reply:
using a constant amount of memory (especially when it's not that much) is often better than dynamically resizing the array. Also with this, unless we statically size the entire array it's almost certain that when we index it we hit the hashmap part, not the array part.